packages/coinstac-ui/app/render/state/ducks/consortia.js

```diff
@@ -1,64 +1,218 @@
+        if (consortium.users.indexOf(username) > -1) {
+          throw new Error(
+            `User ${username} already in consortium ${consortiumId}`
+          );
+        }
+
+        // TODO: Array#push doesn't work.
+        // https://github.com/electron/electron/issues/6734
+        consortium.users = consortium.users.concat(username);
+
+        return consortia.save(consortium);
+      });
   };
 });
 
 /**
- * efficiently update consortia state given a set of consortia which may
- * be new or updated.
- * @param {object} opts
- * @param {function} dispatch redux dispatcher
- * @param {object|object[]} toUpdate POJO consortium/tia to patch onto existing state
- * @param {boolean} [isBg] indicates that this patch has originated from bg-service (vs. user)
- * @returns {undefined}
+ * Set active computation on a consortium.
+ *
+ * @param {string} consortiumId
+ * @param {string} computationId Computation's ID to set as
+ * `activeComputationId` on the consortium model
+ * @returns {Function}
  */
-export const updateConsortia = ({ dispatch, toUpdate, isBg }) => {
-  const currTia = getStore().getState().consortia || [];
-  if (!Array.isArray(toUpdate)) {
-    toUpdate = [toUpdate];
+export const setActiveComputation = applyAsyncLoading(
+  (consortiumId, computationId) => {
+    return () => {
+      const { core: { consortia } } = app;
+
+      if (!computationId) {
+        return Promise.reject('No computation ID specified');
+      }
+
+      return consortia.get(consortiumId)
+        .then(consortium => {
+          consortium.activeComputationId = computationId;
+
+          return consortia.save(consortium);
+        });
+    };
   }
-  toUpdate.forEach((changed) => {
-    const toSwapNdx = findIndex(currTia, { _id: changed._id });
-    if (toSwapNdx >= 0) {
-      currTia[toSwapNdx] = changed;
-    } else {
-      currTia.push(changed);
+);
+
+/**
+ * Remove a user from a consortium.
+ *
+ * Saving the consortium causes the change to flow through `setConsortium`,
+ * updating the state.
+ *
+ * @param {string} consortiumId
+ * @param {string} username
+ * @returns {Function}
+ */
+export const leaveConsortium = applyAsyncLoading((consortiumId, username) => {
+  return () => {
+    const { core: { consortia } } = app;
+
+    return consortia.get(consortiumId)
+      .then(consortium => {
+        const index = consortium.users.indexOf(username);
+
+        if (index < 0) {
+          throw new Error(`User ${username} not in consortium ${consortiumId}`);
+        }
+
+        // TODO: Array#splice doesn't work.
+        // https://github.com/electron/electron/issues/6734
+        consortium.users = consortium.users.filter(u => u !== username);
+
+        return consortia.save(consortium);
+      });
+  };
+});
+
+/**
+ * Update consortia from database change objects.
+ *
+ * @param {Object|Object[]} toUpdate POJO consortium/tia to patch onto existing state
+ * @returns {Function}
+ */
+export function updateConsortia(consortia) {
```

Comment:
Signature change. `isBg` wasn’t doing anything, and we have access do `dispatch` via the thunk middleware. I also moved the new/update logic to the reducer as state modification seems more appropriate there.